python/dist/BUILD

```diff
@@ -0,0 +1,155 @@
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL Google LLC BE LIABLE FOR ANY
+# DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+# ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+# begin:github_only
+#load("//bazel:py_proto_library.bzl", "py_proto_library")
+load(":dist.bzl", "py_dist_module", "py_dist")
+load("@bazel_skylib//rules:common_settings.bzl", "string_flag")
+load("@com_google_protobuf//:protobuf_version.bzl", "PROTOBUF_VERSION")
+load("@rules_python//python:packaging.bzl", "py_wheel")
+
+licenses(["notice"])
+
+py_dist_module(
+    name = "message_mod",
+    module_name = "google.protobuf.pyext._message",
+    extension = "//python:_message",
+)
+
+py_dist_module(
+    name = "api_implementation_mod",
+    module_name = "google.protobuf.internal.api_implementation",
+    extension = "//python:_api_implementation",
+)
+
+# py_proto_library(
+#     name = "well_known_proto_py_pb2",
+#     deps = [
+#         "@com_google_protobuf//:any_proto",
+#         "@com_google_protobuf//:api_proto",
+#         "@com_google_protobuf//:compiler_plugin_proto",
+#         "@com_google_protobuf//:descriptor_proto",
+#         "@com_google_protobuf//:duration_proto",
+#         "@com_google_protobuf//:empty_proto",
+#         "@com_google_protobuf//:field_mask_proto",
+#         "@com_google_protobuf//:source_context_proto",
+#         "@com_google_protobuf//:struct_proto",
+#         "@com_google_protobuf//:timestamp_proto",
+#         "@com_google_protobuf//:type_proto",
+#         "@com_google_protobuf//:wrappers_proto",
+#     ],
+# )
+
+config_setting(
+    name = "aarch64_cpu",
+    values = {"cpu": "linux-aarch_64"},
+)
+
+config_setting(
+    name = "x86_64_cpu",
+    values = {"cpu": "linux-x86_64"},
+)
+
+config_setting(
+    name = "osx-x86_64_cpu",
+    values = {"cpu": "osx-x86_64"},
+)
+
+config_setting(
+    name = "win32_cpu",
+    values = {"cpu": "win32"},
+)
+
+config_setting(
+    name = "win64_cpu",
+    values = {"cpu": "win64"},
+)
+
+py_wheel(
+    name = "binary_wheel",
+    abi = "abi3",
+    distribution = "protobuf",
+    # TODO(https://github.com/protocolbuffers/upb/issues/502): we need to make
+    # this a select() that is calculated from the platform we are actually
+    # building on.
+    platform = select({
+        ":x86_64_cpu": "manylinux2014_x86_64",
+        ":aarch64_cpu": "manylinux2014_aarch64",
+        ":osx-x86_64_cpu": "macosx_12_1_universal",
```

Comment:
I think we'll want to go for 10_9 here to match our existing release: https://pypi.org/project/protobuf/#files

I think this may involve setting MACOSX_DEPLOYMENT_TARGET=10.9. I only know a little about this flag so you'll want to double-check me on that.

Reply:
Edited the name, but leaving this open to talk more about the MACOSX_DEPLOYMENT_TARGET. 

I found a place in the APK files where this variable is set, should I change it there or is there a compiler flag I can change?

Reply:
We should set `MACOSX_DEPLOYMENT_TARGET=10.9` as an environment variable.

Here is some info I found: https://stackoverflow.com/a/33308902

Looks like the easiest way to set it is via `--action_env`: https://stackoverflow.com/a/57902656.

Reply:
Ok that should work! I think that will fit into the transitions part of the work, so I will leave this as is and put a note in the description.